[rfxcom] Sending command to misconfigured device renders RFXCOM bridge unresponsive

Device configuration for a RFY device which worked fine in the march(?) version of openhab1 causes an error in the rfxcom transceiver which returns a message the binding cannot understand. This renders the binding "dead", silently not *transmitting* new commands to the transceiver (receiving seems to continue working)



## Expected Behavior
The bridge should continue dispatch new messages to the transceiver

## Current Behavior
No more messages can be sent

## Log

Init of bridge:
```
2017-06-24 10:01:42.484 [DEBUG] [g.rfxcom.handler.RFXComBridgeHandler] - Message received: Raw data = 140100010253100008060001031C04524658434F4D, Packet type = INTERFACE_MESSAGE, Seq number = 1, Sub type = RESPONSE, Command = GET_STATUS, Transceiver type = _433_92MHZ_TRANSCEIVER, Hardware version = 0.1, Firmware type = unknown, Firmware version = 16, Output power = 0dBm, Undecoded packets = false, RFU6 packets = false, Byron SX packets packets (433.92) = false, RSL packets packets (433.92) = false, Lighting4 packets (433.92) = false, FineOffset / Viking (433.92) packets = false, Rubicson (433.92) packets = false, AE (433.92) packets = false, BlindsT1/T2/T3 (433.92) packets = false, BlindsT0 (433.92) packets = false, ProGuard (868.35 FSK) packets = false, FS20/Legrand CAD (868.35/433.92) packets = false, La Crosse (433.92/868.30) packets = true, Hideki/UPM (433.92) packets = false, AD LightwaveRF (433.92) packets = false, Mertik (433.92) packets = false, Visonic (315/868.95) packets = false, ATI (433.92) packets = false, Oregon Scientific (433.92) packets = false, Meiantech (433.92) packets = false, HomeEasy EU (433.92) packets = false, AC (433.92) packets = true, ARC (433.92) packets = true, X10 (310/433.92) packets = false, HomeConfort (433.92) packets = false, KEELOQ (433.92/868.95) packets = false
2017-06-24 10:01:42.501 [DEBUG] [g.rfxcom.handler.RFXComBridgeHandler] - RFXCOM transceiver/receiver type: _433_92MHZ_TRANSCEIVER, hw version: 0.1, fw version: 16
2017-06-24 10:01:42.505 [DEBUG] [g.rfxcom.handler.RFXComBridgeHandler] - Ignoring transceiver configuration
2017-06-24 10:01:42.507 [DEBUG] [g.rfxcom.handler.RFXComBridgeHandler] - Start receiver
```

Sending command to shutter:

```
2017-06-24 10:05:51.427 [DEBUG] [g.rfxcom.handler.RFXComBridgeHandler] - Transmitting message 'Raw data = unknown, Packet type = RFY, Seq number = 0, Sub type = RFY, Unit Id = 101011.1, Unit Code = 1, Command = DOWN, Signal level = 0'
2017-06-24 10:05:51.429 [TRACE] [rnal.connector.RFXComSerialConnector] - Send data (len=13): 0C1A0000018A93010300000000

==> /var/log/openhab2/events.log <==
2017-06-24 10:05:51.451 [ItemCommandEvent          ] - Item 'Shutter_gHovedetasje_Stue_Sofa' received command DOWN
2017-06-24 10:05:51.468 [ItemStateChangedEvent     ] - Shutter_gHovedetasje_Stue_Sofa changed from 0.0 to 100

==> /var/log/openhab2/openhab.log <==
2017-06-24 10:05:51.635 [ERROR] [g.rfxcom.handler.RFXComBridgeHandler] - Error occurred during packet receiving, data: 140101000153100008060001031C04524658434F4D
org.openhab.binding.rfxcom.internal.exceptions.RFXComUnsupportedValueException: Unsupported value '1' for Commands
	at org.openhab.binding.rfxcom.internal.messages.RFXComInterfaceMessage$Commands.fromByte(RFXComInterfaceMessage.java:92)
	at org.openhab.binding.rfxcom.internal.messages.RFXComInterfaceMessage.encodeMessage(RFXComInterfaceMessage.java:271)[205:org.openhab.binding.rfxcom:2.1.0.201706222153]
	at org.openhab.binding.rfxcom.internal.messages.RFXComInterfaceMessage.<init>(RFXComInterfaceMessage.java:211)[205:org.openhab.binding.rfxcom:2.1.0.201706222153]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)[:1.8.0_131]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)[:1.8.0_131]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)[:1.8.0_131]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)[:1.8.0_131]
	at org.openhab.binding.rfxcom.internal.messages.RFXComMessageFactory.createMessage(RFXComMessageFactory.java:130)[205:org.openhab.binding.rfxcom:2.1.0.201706222153]
	at org.openhab.binding.rfxcom.handler.RFXComBridgeHandler$MessageListener.packetReceived(RFXComBridgeHandler.java:233)[205:org.openhab.binding.rfxcom:2.1.0.201706222153]
	at org.openhab.binding.rfxcom.internal.connector.RFXComBaseConnector.sendMsgToListeners(RFXComBaseConnector.java:42)[205:org.openhab.binding.rfxcom:2.1.0.201706222153]
	at org.openhab.binding.rfxcom.internal.connector.RFXComStreamReader.run(RFXComStreamReader.java:73)[205:org.openhab.binding.rfxcom:2.1.0.201706222153]
```


<!-- If describing a bug, tell us what happens instead of the expected behavior -->
<!-- Include related log information (preferably debug level) and related configs. -->
<!-- Use file attachment for log and config information longer than a few lines -->
<!-- If suggesting a change/improvement, explain the difference from current behavior -->
<!-- For improvements, discuss at community.openhab.org first and include link to topic -->

## Possible Solution
Add support for command response type '1'
<!-- Not obligatory, but suggest a fix/reason for the bug, -->
<!-- or ideas how to implement the addition or change -->

## Steps to Reproduce (for Bugs)
Add rfxcom item in .item file:
```
Rollershutter Shutter_gHovedetasje_Stue_Sofa "Screen sofa" <rollershutter> ( gRestoreOnStartup,gShutterStue)  {channel="rfxcom:rfy:usb0:Shutter_gHovedetasje_Stue_Sofa:shutter"}
```

Add thing in .thing file:

```
Bridge rfxcom:bridge:usb0 [ serialPort="/dev/ttyUSB0" ] {
	Thing rfy Shutter_gHovedetasje_Stue_Sofa "Shutter_gHovedetasje_Stue_Sofa" [deviceId="101011.1", subType="RFY"]
}
```



## Context
Trying to make a SOMFY shutter work in Openhab2 based on Openhab1 configuration details
<!-- How has this issue affected you? What are you trying to accomplish? -->
<!-- Providing context helps us come up with a solution that is most useful in the real world -->



## Your Environment
RPI2. 
MyOpenhabian image
Openhab2 snapshot as of 2017-06-14.

--- a/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java
+++ b/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java
@@ -274,11 +274,13 @@ public class RFXComBridgeHandler extends BaseBridgeHandler {
                             logger.debug("Start receiver");
                             connector.sendMessage(RFXComMessageFactory.CMD_START_RECEIVER);
                         }
-                    } else if (msg.subType == SubType.START_RECEIVER) {
-                        logger.debug("Start TX of any queued messages");
-                        transmitQueue.send();
+                    } else {
+                        if (msg.subType == SubType.START_RECEIVER) {
+                            updateStatus(ThingStatus.ONLINE);
+                            logger.debug("Start TX of any queued messages");
+                        }
 
-                        updateStatus(ThingStatus.ONLINE);
+                        transmitQueue.send();
                     }
                 } else if (message instanceof RFXComTransmitterMessage) {
                     RFXComTransmitterMessage resp = (RFXComTransmitterMessage) message;

--- a/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java
+++ b/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java
@@ -274,13 +274,13 @@ public class RFXComBridgeHandler extends BaseBridgeHandler {
                             logger.debug("Start receiver");
                             connector.sendMessage(RFXComMessageFactory.CMD_START_RECEIVER);
                         }
-                    } else {
-                        if (msg.subType == SubType.START_RECEIVER) {
-                            updateStatus(ThingStatus.ONLINE);
-                            logger.debug("Start TX of any queued messages");
-                        }
-
+                    } else if (msg.subType == SubType.START_RECEIVER) {
+                        updateStatus(ThingStatus.ONLINE);
+                        logger.debug("Start TX of any queued messages");
                         transmitQueue.send();
+                    } else {
+                        logger.debug("Interface response received: {}", msg);
+                        transmitQueue.sendNext();
                     }
                 } else if (message instanceof RFXComTransmitterMessage) {
                     RFXComTransmitterMessage resp = (RFXComTransmitterMessage) message;

--- a/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComInterfaceMessage.java
+++ b/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComInterfaceMessage.java
@@ -268,9 +268,9 @@ public class RFXComInterfaceMessage extends RFXComBaseMessage {
         super.encodeMessage(data);
 
         subType = SubType.fromByte(super.subType);
-        command = Commands.fromByte(data[4]);
 
         if (subType == SubType.RESPONSE) {
+            command = Commands.fromByte(data[4]);
             transceiverType = TransceiverType.fromByte(data[5]);
 
             firmwareVersion = data[6] & 0xFF;
@@ -327,6 +327,8 @@ public class RFXComInterfaceMessage extends RFXComBaseMessage {
             text = "";
 
         } else if (subType == SubType.START_RECEIVER) {
+            command = Commands.fromByte(data[4]);
+
             final int len = 16;
             final int dataOffset = 5;
 
@@ -341,6 +343,11 @@ public class RFXComInterfaceMessage extends RFXComBaseMessage {
             } catch (UnsupportedEncodingException e) {
                 // ignore
             }
+        } else {
+            // We don't handle the other subTypes but to avoid null pointer
+            // exceptions we set command to something. It doesn't really
+            // matter what but it may b printed in log messages so...
+            command = Commands.UNSUPPORTED_COMMAND;
         }
     }
 

